structure/hash.go

```diff
@@ -105,8 +105,9 @@ func (t *TxStructure) updateHash(key []byte, field []byte, fn func(oldValue []by
 		return errors.Trace(err)
 	}
 
+	newMeta := meta
 	if oldValue == nil {
-		meta.Length++
+		newMeta.Length++
```

Comment:
if oldValue is nil and newValue is nil too, I don't think we need add length and can return directly.